pkg/utils/k8sutils.go

```diff
@@ -114,6 +114,16 @@ func AddAnnotation(obj metav1.Object, key string, value string) bool {
 	return false
 }
 
+// AddAnnotations adds values from newAnnotations to the keys on the supplied obj or overwrites values for existing keys on the obj
+func AddAnnotations(obj metav1.Object, newAnnotations map[string]string) {
```

Comment:
we have two implementations to add annotations AddAnnotations and AddAnnotation, cannot we use the exsting one? Do we have a need to have two functions to add annotations

Reply:
Same thing as done with labels. I guess it's just a byproduct of the need to set a single annotation from here and a list of such that come from the server side.